pkg/internal/controller/controller_test.go

```diff
@@ -169,6 +172,108 @@ var _ = Describe("controller", func() {
 			close(done)
 		}, 10.0)
 
+		It("should not error when source.channel is passed", func(done Done) {
```

Comment:
I would probably rename this to `should process events from source.Channel`. While it is possible that `source.Channel` could error and that would fail the test, the main failure we are testing for here will manifest as the test timing out and if someone doesn't know the test they'd be like "Well it doesn't error, why is it failing anyways?"